site/docs/plugins/telegram-entities-parser.md

```diff
@@ -0,0 +1,247 @@
+---
+prev: false
+next: false
+---
+
+# Convert Telegram Entities (telegram-entities-parser)
```

Comment:
Not sure if this is the write place to suggest this, but I think this plugin is better be called entity parser.

Reply:
Do you mean 'entity-parser' as the plugin name or the package name? If you’re referring to the plugin, I agree with you.

I’m not sure if I’m correct, but should 'entity' be plural instead of singular in this context?

Reply:
Both the plugin name and the package name. No, it should be singular.

Reply:
PTAL https://github.com/grammyjs/website/pull/1121/commits/1f4bbdae87d2d1bc6cd68511803c21aa70f19663.

Reply:
Do you also want to rename `@qz/telegram-entities-parser`?

Reply:
I think it's best to keep the actual package name as `telegram-entities-parser`. The name `entity-parser` might be suitable for grammY documentation since it relates to Telegram, but it could be confusing outside of grammY.

Reply:
Good idea. I still think `@qz/telegram-entity-parser` is better. What about you?

Also, shouldn't we publish it under grammyjs on JSR in your opinion, @KnorpelSenf?

Reply:
> Also, shouldn't we publish it under grammyjs on JSR in your opinion, @KnorpelSenf?

No. When people want to do it the grammY way, they use entities directly. This avoids an entire class of errors, which is much better than causing them and then fixing them. However, not everybody wants to listen to us, and people use Markdown/HTML anyway. Also, entities fall short when it comes to the i18n plugin. That's why it's great not to leave people hanging, and provide a third-party plugin.

Reply:
> Good idea. I still think @qz/telegram-entity-parser is better. What about you?

Sounds good to me. I can change the package and repo names, but JSR doesn’t allow changes unless I create a new one. I previously renamed it from `grammy-entity`, which led to an abandoned package here: https://jsr.io/@qz/grammy-entity. Renaming again might clutter JSR, so it’s best to keep the current name until we find a cleaner way to change it.

> No. When people want to do it the grammY way, they use entities directly. This avoids an entire class of errors, which is much better than causing them and then fixing them. However, not everybody wants to listen to us, and people use Markdown/HTML anyway.

Agree. I've added clear warnings in the documentation that this plugin might not be the best choice. I also included a section on handling common issues without this package. Personally, I can’t work with entities directly since I need to convert them to HTML for the website, so I created this plugin to help others in similar situations.